### PR TITLE
fix(specs): usage api

### PR DESCRIPTION
**Pull request by kai687**
## 🧭 What and Why

Improve the Usage API spec.

🎟 JIRA Ticket: [DOC-1499](https://algolia.atlassian.net/browse/DOC-1499)

### Changes included:

- Remove (internal) `usage-dev` server from the spec
- Add `usage` tag to each method (tag was defined in the spec, but not added to the methods)
- Start date and end date must be full RFC 3339 dates

[DOC-1499]: https://algolia.atlassian.net/browse/DOC-1499?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by shortcuts — DISMISSED** [review]
looks great!

File: specs/usage/paths/statisticIndex.yml [hunk 1]
```diff
@@ -3,6 +3,8 @@ get:
   summary: Retrieve usage information for one index
   description: |
     Retrieves the selected usage statistics for one index.
+  tags:
+    - usage
```
Comment (millotp) [inline specs/usage/paths/statisticIndex.yml:7]: `Error: Tag name "usage" must be different from client name usage in operation getUsage`

not sure why there is a clash

Comment (shortcuts) [reply specs/usage/paths/statisticIndex.yml:7]: that might be a documentation requirement, I can't remember

**Review by shortcuts — APPROVED** [review]
gg!

**Pull request merged**